ccpp/data/GFS_typedefs.F90

```diff
@@ -160,6 +160,7 @@ module GFS_typedefs
     real (kind=kind_phys), pointer :: pgr  (:)     => null()  !< surface pressure (Pa) real
     real (kind=kind_phys), pointer :: ugrs (:,:)   => null()  !< u component of layer wind
     real (kind=kind_phys), pointer :: vgrs (:,:)   => null()  !< v component of layer wind
+    real (kind=kind_phys), pointer :: wgrs (:,:)   => null()  !< w component of layer wind
```

Comment:
Even though it's a bit more computation, I would strongly suggest just passing in omega to maximum_hourly_diagnostics rather than trying to use `w` from the dycore. It's much simpler (you don't have to worry about the logic around whether wgrs should exist and adding a dependency of this diagnostic on whether the dycore is run hydrostatically or not). There is precedence: w is calculated from omega in the following schemes: GFDL MP, MG MP, NSSL MP, Thompson MP, MYNNEDMF PBL, SHOC. If you wanted to get fancy, you could pass the calculated w out of the microphysics schemes (or otherwise check whether w has already been calculated) within maximum_hourly_diagnostics to avoid recalculating if you're really worried about avoiding duplicate calculation.

Reply:
Re-posting @welewis69's response to a similar comment:

> Chunxi and others, our only concern regarding using wgrs as solved for from vvl = dp/dz*wgrs would be that the lightning threat indices are based on empirical relationships derived using the non-hydrostatic vertical velocity. In FV3, is the fully non-hydrostatic w recoverable from vvl in this manner?

Reply:
Thanks much for the discussion and clarification of the various issues, everyone. Two of the three indices depend on graupel fluxes, and since these fluxes tend to be concentrated in the updrafts, we really do need to use the w field without smoothing if at all possible (otherwise, we will produce fields that are simultaneously too broad and too weak).